### PR TITLE
Use ChildStdout instead of Stdio

### LINKED ISSUE #72: Change pipeline to use ChildStdout instead of Stdio

### PR DESCRIPTION
**Pull request by siliconlad**
image

### CODE REVIEW

File: src/redirect.rs [hunk 1]
```diff
@@ -163,7 +164,12 @@ impl Runnable for InputRedirect {
     fn run(&self) -> Result<String, Box<dyn Error>> {
         let file = File::open(&self.input_file)?;
         let file_fd = file.into_raw_fd();
-        let input = unsafe { Stdio::from_raw_fd(file_fd) };
+
+        let mut child = Command::new("cat")
+            .stdin(unsafe { Stdio::from_raw_fd(file_fd) })
+            .stdout(Stdio::piped())
+            .spawn()?;
+        let input = child.stdout.take().unwrap();
```
Comment (siliconlad) [inline src/redirect.rs:172]: This (and what I did below) is a bit of a hack :(

File: src/pipeline.rs [hunk 2]
```diff
@@ -23,13 +23,13 @@ impl Pipeline {
...
         for (i, command) in self.commands.iter().enumerate() {
-            let prev_stdout_ = match prev_stdout.take() {
-                Some(stdout) => stdout,
-                None => Stdio::null(),
-            };
-            let cmd_stdout = command.pipe(Some(prev_stdout_))?;
+            // let prev_stdout_ = match prev_stdout.take() {
+            //     Some(stdout) => Stdio::from(stdout),
+            //     None => Stdio::null(),
+            // };
```
Comment (CraigNewlands) [inline src/pipeline.rs]: Can delete these lines

File: src/redirect.rs (hunk 1 above)
Comment (CraigNewlands) [reply src/redirect.rs:172]: Sad but needs must :(

File: src/pipeline.rs (hunk 2 above)
Comment (siliconlad) [reply src/pipeline.rs]: Done

**Review by CraigNewlands — APPROVED** [review]
:rocket:

**Pull request merged**